### ISSUE TITLE
/ info endpoint doesn't return 2FA flag for authenticated users with 2FA disabled

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/api#1380]
When I login with a user that has no 2FA setup in a role that has 2FA enforced, the API doesn't return the 2FA required flag, making it impossible for the app to boot someone to the setup-2fa page.

* Authentication goes through like normal 
* `/_/` doesn't return the 2fa required flag:

```
{"data":{"api":{"project_color":"pink-500","project_name":"Rijk's Local","project_image":null,"project_icon":null,"project_logo":""}},"public":true}
```

**Comment by bjgajjar** [comment]
Hi @rijkvanzanten,

When 2FA is not enabled then the system will throw the exception and not allow that user to log in.  Thus, API will not have an ID of login user and the cause of that system can't find out whether the 2FA is enabled for a user or not. 

This issue may occur when user is already logged in and meanwhile the admin enforces the 2FA. For that, we can set the `required2FA` flag as a `false` by default. 

   - Authenticated User 
       - Return `true` if the user enabled the 2FA
       - Return `false` if the user did not enable the 2FA
   - Unauthenticated User (`Public` Role)
       - Return `false`

So basically, we can achieve it like; the endpoint will return `required2FA` every time whether the user is logged in or not.

And it will not affect the system I guess. Please point out if I am missing any points.

**Comment by rijkvanzanten** [comment]
This doesn't sound right. Currently, a user is always able to log in. Even when 2FA is required but not enabled. The only thing the user can do in that case is update /users/me to set the 2FA token

**Comment by bjgajjar** [comment]
> This doesn't sound right. Currently, a user is always able to log in.

Yeah; I agree. The API will provide you the token for `/login` endpoint, but it will not add that user to ACL. 

The authentication middleware will authenticate the authentication token which is set in the request header and add that user in `ACL`; if the 2FA is enforced and the user didn't enable that; the exception will be thrown. On the catch of that exception; the system will set `public` permission if the `public` role exists otherwise the exception will be thrown. 

In our case, the same scenarios occur. Unless the user is not fetched from ACL; `/_/` endpoint will not provide some details [`version`, `requires2FA`, `max_upload_size`, `database`]. 

Thus I set the `requires2FA` as a default value in #1388. But the drawback is; although the authentication token is not set in request header; this flag will be returned.

**Comment by rijkvanzanten** [comment]
This information is returned in `/auth/authenticate` now
